django-backend/scripts/generate_transaction_record_fixture.py

```diff
@@ -0,0 +1,269 @@
+    )
+    parser.add_argument(
+        '--report_b_id',
+        default="1406535e-f99f-42c4-97a8-247904b7d297",
+        help='secondary report (default: 1406535e-f99f-42c4-97a8-247904b7d297)'
+    )
+    parser.add_argument(
+        '--report_start_date',
+        default="2005-01-30",
+        help='default value: 2005-01-30'
+    )
+    parser.add_argument(
+        '--report_end_date',
+        default="2005-02-28",
+        help='default value: 2005-02-28'
+    )
+    parser.add_argument(
+        '--output_file',
+        help="Output to a file instead of the console (e.g --output_file=records.txt)")
+
+    return parser.parse_args()
+
+
+def get_schedule(form_type):
+    schedule = str(form_type[1]).capitalize()  # the second char in the form_type
+    if schedule == "C" and form_type[2] in ["1", "2"]:
+        schedule = form_type[1:2]
+    return schedule
+
+
+def get_comment(form_type, test_case):
+    if test_case == "within_dates":
+        return f"{form_type} within report dates, should count in Col. A"
+    if test_case == "within_year":
+        return f"{form_type} within year, should count in Col. B"
+    if test_case == "outside_dates":
+        return f"{form_type} outside of dates, should not count"
+    if test_case == "within_dates_but_memo":
+        return f"{form_type} is a memo item, should not count"
+    return ""
+
+
+def get_amount_field(schedule):
+    if schedule == "A":
+        return "contribution_amount"
+    if schedule == "B":
+        return "expenditure_amount"
+    if schedule == "C":
+        return "loan_amount"
+    if schedule == "C1":
+        return "loan_amount"
+    if schedule == "C2":
+        return "guaranteed_amount"
+    if schedule == "D":
+        return "incurred_amount"
+    if schedule == "E":
+        return "expenditure_amount"
+    return ""
+
+
+def get_date_field(schedule):
+    if schedule == "A":
+        return "contribution_date"
+    if schedule == "B":
+        return "expenditure_date"
+    if schedule == "C":
+        return "loan_incurred_date"
+    if schedule == "C1":
+        return "loan_incurred_date"
+    if schedule == "C2":
+        return ""
+    if schedule == "D":
+        return ""
+    if schedule == "E":
+        return "disbursement_date"
+    return ""
+
+
+def get_date(args, test_case):
+    report_start = args.report_start_date.split("-")
+    report_end = args.report_end_date.split("-")
+
+    if "within_dates" in test_case:
+        return "-".join(report_start)
+
+    if "year" in test_case:
+        months = set([str(m).rjust(2, "0") for m in range(1, 13)])
+        report_start_month = report_start[1]
+        report_months = range(int(report_start_month), 13)
+        formatted_report_months = set([str(m).rjust(2, "0") for m in report_months])
+        months_before_report = list(months - formatted_report_months)
+        if len(months_before_report) > 0:
+            random_month = choice(months_before_report)
```

Comment:
No need to change this, but a downside of making this non-deterministic is that running the script again will produce a different output.